internal/app/config.go

```diff
@@ -218,6 +220,18 @@ func validateOutputConfig(outputs []ConfigOutput) []error {
 		if output.CloudStorage.Bucket != "" && output.CloudStorage.Path == "" {
 			errors = append(errors, fmt.Errorf("invalid output - path empty for bucket: %s", output.CloudStorage.Bucket))
 		}
+		errors = append(errors, validatePubSubConfig(output.PubSub)...)
+	}
+	return errors
+}
+
+func validatePubSubConfig(pubsub PubSubOutput) []error {
+	var errors = make([]error, 0)
+	if pubsub.Project != "" && pubsub.Topic == "" {
+		errors = append(errors, fmt.Errorf("PubSub Topic is not set for the project [%s]", pubsub.Project))
+	}
+	if pubsub.Topic != "" && pubsub.Project == "" {
```

Comment:
I prefer to have a project configuration for Pub/Sub or GCS outputs. A number of customers use separate project for logs (it's a best practice also). We can use the GKE project for outputs if a project is not configured. For example, if Pub/Sub output has topic configured but project is not, then use the Cluster's project. But if multiple projects are configured, then it would be a problem.

Reply:
GCS bucket names are global, they don't require project. And for pub/sub config, having a project makes perfect sense.